test/utils.go

```diff
@@ -0,0 +1,61 @@
+package test
+
+import (
+	"fmt"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"math/rand"
+	"time"
+)
+
+const (
+	charset = "abcdefghijklmnopqrstuvwxyz0123456789"
+)
+
+func NewLogger() (*zap.Logger, error) {
+	loggerConfig := zap.NewDevelopmentConfig()
+	loggerConfig.EncoderConfig.TimeKey = "timestamp"
+	loggerConfig.Encoding = "json"
+	loggerConfig.EncoderConfig.EncodeTime = zapcore.TimeEncoderOfLayout("Jan 02 15:04:05.000000000")
+
+	return loggerConfig.Build()
+}
+
+func NewSugaredLogger() (*zap.SugaredLogger, error) {
+	logger, err := NewLogger()
+	if err != nil {
+		return nil, err
+	}
+	return logger.Sugar(), nil
+}
+
+func NewNamespace(name string) *apiv1.Namespace {
+	namespace := apiv1.Namespace{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Namespace",
+			APIVersion: "v1",
+		},
```

Comment:
(nit) You can remove these lines. The k8s client can figure out the type meta at runtime.